clippy_utils/src/macros.rs

```diff
@@ -765,12 +767,82 @@ pub struct FormatArgsExpn<'tcx> {
     /// Has an added newline due to `println!()`/`writeln!()`/etc. The last format string part will
     /// include this added newline.
     pub newline: bool,
-    /// Values passed after the format string and implicit captures. `[1, z + 2, x]` for
+    /// Spans of the commas between the format string and explicit values, excluding any trailing
+    /// comma
+    ///
+    /// ```ignore
+    /// format!("..", 1, 2, 3,)
+    /// //          ^  ^  ^
+    /// ```
+    comma_spans: Vec<Span>,
+    /// Explicit values passed after the format string, ignoring implicit captures. `[1, z + 2]` for
     /// `format!("{x} {} {y}", 1, z + 2)`.
-    value_args: Vec<&'tcx Expr<'tcx>>,
+    explicit_values: Vec<&'tcx Expr<'tcx>>,
```

Comment:
we may want to clarify this comment -- the above `format!(...)` doesn't look like valid code.

Reply:
Ah whoops, the `{y}` should be `{}` I also forgot to remove [the FIXME](https://github.com/rust-lang/rust-clippy/blob/ac0e10aa68325235069a842f47499852b2dee79e/clippy_lints/src/format_args.rs#L176-L186) if you fancy doing a cleanup PR, otherwise I'll get around to it